### PR TITLE
Fix pensions spouseIsVeteran

### PR DESCRIPTION
**Pull request by TaiWilkin**
## Summary

When the veteran's marital status was set to Separated, the "spouseIsVeteran" field wasn't correctly populated in the PDF. 

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/75004

## Testing done

- [x] New code is covered by unit tests

Steps:

-  Start a new Pension application
- Enter minimal data until you get to the marriage section. Select "Separated" for the marital status
- Enter spouse's into making sure to select YES for the question "Is your spouse a veteran?"
continue with minimal data
- Submit the form and inspect the generated PDF

Before this PR: In the PDF, NO is selected for this question
New result: In the PDF, YES should be selected

## Screenshots

Should populate field if separated:
Screenshot 2024-01-31 at 11 11 45 AM
Should populate field if married:
Screenshot 2024-01-31 at 11 15 26 AM
Should not populate field for other marital statuses:
Screenshot 2024-01-31 at 11 16 50 AM

## What areas of the site does it impact?
Benefits Pensions

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature

### CODE REVIEW

File: spec/fixtures/pdf_fill/21P-527EZ/overflow.json [hunk 1]
```diff
@@ -138,7 +138,7 @@
   ],
   "spouseDateOfBirth": "1960-01-01",
   "spouseSocialSecurityNumber": "333224444",
-  "spouseIsVeteran": true,
+  "spouseIsVeteran": false,
```
Comment (danlim715) [inline spec/fixtures/pdf_fill/21P-527EZ/overflow.json:141]: we should revert this back to true, to stay in sync with the front-end fixtures.

File: spec/fixtures/pdf_fill/21P-527EZ/kitchen_sink.json [hunk 2]
```diff
@@ -100,7 +100,7 @@
   ],
   "spouseDateOfBirth": "1960-01-01",
   "spouseSocialSecurityNumber": "333224444",
-  "spouseIsVeteran": true,
+  "spouseIsVeteran": false,
```
Comment (danlim715) [inline spec/fixtures/pdf_fill/21P-527EZ/kitchen_sink.json:103]: we should revert this back to true, to stay in sync with the front-end fixtures.

Comment (TaiWilkin) [inline spec/fixtures/pdf_fill/21P-527EZ/kitchen_sink.json:103]: Changed this to false to give the currently expected PDF output (I'm not able to update the PDFs at this time). We can flip this and update the PDF when less time-constrained if desired

File: spec/fixtures/pdf_fill/21P-527EZ/merge_fields.json [hunk 3]
```diff
@@ -141,7 +141,7 @@
     "second": "22",
     "third": "4444"
   },
-  "spouseIsVeteran": true,
+  "spouseIsVeteran": 1,
```
Comment (danlim715) [inline spec/fixtures/pdf_fill/21P-527EZ/merge_fields.json:144]: This should be 0 for true.

**Review by danlim715 — CHANGES_REQUESTED** [review]

**Review by danlim715 — APPROVED** [review]

**Review by danlim715 — APPROVED** [review]

**Pull request merged**